File: test_hash.lua
local function domainFNV1a(domain)
    local hash = 0xCBF29CE484222325
    local prime = 0x100000001B3
    for part in string.gmatch(domain, "[^%.]+") do
        hash = hash ~ #part
        hash = hash * prime
        for i=1, #part do
            local byte = string.byte(part, i)
            if byte >= 65 and byte <= 90 then byte = byte + 32 end
            hash = hash ~ byte
            hash = hash * prime
        end
    end
    return hash
end

print(string.format("0x%X", domainFNV1a("youporn.com")))
print(string.format("0x%X", domainFNV1a("salt_byouporn.com")))

File: test_hash.go
package main
import (
	"fmt"
	"strings"
)

var fnvOffsetBasis = uint64(0xcbf29ce484222325)
var fnvPrime       = uint64(0x100000001b3)

func domainFNV1a(domain string) uint64 {
	domain = strings.TrimSuffix(strings.ToLower(domain), ".")
	parts := strings.Split(domain, ".")
	hash := fnvOffsetBasis
	for _, part := range parts {
		hash ^= uint64(len(part))
		hash *= fnvPrime
		for i := 0; i < len(part); i++ {
			c := part[i]
			if c >= 'A' && c <= 'Z' { c += 32 }
			hash ^= uint64(c)
			hash *= fnvPrime
		}
	}
	return hash
}

func main() {
	fmt.Printf("0x%X\n", domainFNV1a("youporn.com"))
	fmt.Printf("0x%X\n", domainFNV1a("salt_byouporn.com"))
}
